qemu/tests/blockdev_inc_backup_xpt_incon_bitmap.py

```diff
@@ -0,0 +1,159 @@
+            src_img = self.source_disk_define_by_params(self.params,
+                                                        self.src_img_tags[0])
+            output = json.loads(src_img.info(output="json"))
+            bitmaps = output["format-specific"]["data"].get("bitmaps")
+            if bitmaps:
+                for bitmap in bitmaps:
+                    if bitmap["name"] == bitmap_name:
+                        return bitmap
+            return None
+
+        bitmap_info = _get_bitmap_info(self.bitmaps[0])
+        if not bitmap_info:
+            self.test.fail("Persistent bitmap not stored in image")
+        if inconsistent and "in-use" not in bitmap_info["flags"]:
+            self.test.fail("Bitmap stored is not inconsistent")
+
+    def expose_inconsistent_bitmap(self):
+        def _nbd_expose_cmd(qemu_nbd, filename, local_image, params):
+            cmd_dict = {
+                "export_format": "",
+                "persistent": "-t",
+                "port": "",
+                "filename": "",
+                "fork": "--fork",
+                "pid_file": "",
+                "bitmap": "",
+                }
+            export_cmd = ('{export_format} {persistent} {port} {bitmap} '
+                          '{fork} {pid_file} {filename}')
+            pid_file = utils_misc.generate_tmp_file_name('%s_nbd_server'
+                                                         % local_image, 'pid')
+            cmd_dict['pid_file'] = '--pid-file %s' % pid_file
+            cmd_dict['filename'] = filename
+            if params.get('nbd_export_format'):
+                cmd_dict['export_format'] = '-f %s' % params['nbd_export_format']
+            else:
+                if params.get('nbd_port'):
+                    cmd_dict['port'] = '-p %s' % params['nbd_port']
+            if params.get('nbd_export_bitmaps'):
+                cmd_dict['bitmap'] = "".join(
+                    [" -B %s" % _ for _ in params['nbd_export_bitmaps'].split()])
+            cmdline = qemu_nbd + ' ' + string.Formatter().format(export_cmd,
+                                                                 **cmd_dict)
+            return cmdline
+
+        logging.info("Export inconsistent bitmap with qemu-nbd")
+        cmd = _nbd_expose_cmd(self.nbd_exports[0]._qemu_nbd,
+                              self.nbd_exports[0]._local_filename,
+                              self.nbd_exports[0]._tag,
+                              self.nbd_exports[0]._image_params)
+        result = process.run(cmd, ignore_status=True, shell=True,
+                             ignore_bg_processes=True)
+        error_msg = self.params.get("error_msg") % self.bitmaps[0]
+        if error_msg not in result.stderr.decode():
+            self.test.fail(result.stderr.decode())
+
+    def do_test(self):
+        self.do_full_backup()
+        self.gen_inc_files()
+        self.main_vm.destroy(free_mac_addresses=False)
+        self.check_bitmap_status()
```

Comment:
Do we need these two steps? After file I/O, we kill qemu-kvm, then we should get an inconsistent bitmap

Reply:
add persistent bimap->shutdown vm    -------> Persistent bitmap can be saved on image
-->restart vm --> kill it                           ----> Persistent bitmap will become inconsistent.


If we just add persistent bitmap --> kill vm   ---->Bitmap info won't stored on image and no inconsistent bitmap will be produce.